server.go

```diff
@@ -529,6 +529,7 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 		NoAnchors:                cfg.ProtocolOptions.NoAnchorCommitments(),
 		NoWumbo:                  !cfg.ProtocolOptions.Wumbo(),
 		NoScriptEnforcementLease: cfg.ProtocolOptions.NoScriptEnforcementLease(),
+		NoKeysend:                !cfg.AcceptKeySend,
```

Comment:
i've opted to just using the existing config option rather than moving it to the `ProtocolOptions` struct since that would break the API which i dont think is worth it for this change.